src/careamics/dataset_ng/patch_extractor/image_stack/czi_image_stack.py

```diff
@@ -0,0 +1,297 @@
+import re
+from collections.abc import Sequence
+from pathlib import Path
+from typing import Any, Literal
+
+import numpy as np
+from numpy.typing import NDArray
+from pylibCZIrw.czi import CziReader, Rectangle, open_czi
+
+
+class CziImageStack:
+    """
+    A class for extracting patches from an image stack that is stored as a CZI file.
+
+    Parameters
+    ----------
+    data_path : str or Path
+        Path to the CZI file.
+
+    scene : int, optional
+        Index of the scene to extract. If not specified, the entire image covering
+        all scenes present in the CZI file will be extracted.
+        The scene can also be provided as part of `data_path` by appending an `"@"`
+        followed by the scene index to the filename.
+
+    depth_axis : {"none", "Z", "T", "auto"}, default: "auto"
+        Which axis to use as depth-axis for providing 3-D patches.
+
+        - `"none"`: Only provide 2-D patches. If a Z or T dimension is present in the
+          data, they will be combined into the sample dimension `S`.
+        - `"Z"`: Use the Z-axis as depth-axis. If a T axis is present as well, it will
```

Comment:
Are both `auto` and `Z` necessary? `auto` would always select `Z` if it is present, so it would be equivalent.

Reply:
See the other discussion, as the two comments were linked!